src/components/SharedComponents/LocationDisplay.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+
+import { Body4 } from "components/SharedComponents";
+import { View } from "components/styledComponents";
+import * as React from "react";
+import { useTranslation } from "react-i18next";
+import IconMaterial from "react-native-vector-icons/MaterialIcons";
+import useLocationName from "sharedHooks/useLocationName";
+
+type Props = {
+  latitude: number,
+  longitude: number
+};
+
+const LocationDisplay = ( {
+  latitude,
+  longitude
+}: Props ): React.Node => {
+  const { t } = useTranslation( );
+  const locationName = useLocationName( latitude, longitude );
+
+  return (
+    <View className="flex flex-row items-center">
+      <IconMaterial name="location-pin" size={15} />
+      <Body4 className="text-darkGray ml-[5px]">
+        {locationName || t( "no place guess" )}
```

Comment:
Please change "no place guess" to "Missing Location" and [add it to the translations file](https://github.com/inaturalist/iNaturalistReactNative#adding-new-text) (FYI, the capitalization of the key should match the text if possible, so here it would be `Missing-Location`).